src/settings/connectionsettings.ts

```diff
@@ -372,7 +372,8 @@ export function isSonarQubeConnection(connection: BaseConnection): connection is
 }
 
 function getTokenStorageKey(connection: SonarQubeConnection | SonarCloudConnection) {
-  return isSonarQubeConnection(connection) ? connection.serverUrl : connection.organizationKey;
+  const regionPrefix = !isSonarQubeConnection(connection) && connection.region ? `${connection.region}_` : '';
```

Comment:
this is for legacy connections who might not have region specifically set.

Reply:
This will not work for the (arguably very few) people who already have a connection with a region: the existing token will not be found.

Note that it may be acceptable to have these users re-configure their token in this case.